fab/operations/release.py

```diff
@@ -66,6 +48,87 @@ def update_code(git_tag, use_current_release=False):
         sudo("find . -name '*.pyc' -delete")
 
 
+def create_offline_dir():
+    run('mkdir -p {}'.format(env.offline_code_dir))
+
+
+def update_code_offline():
+    clone_current_release_to_home_directory()
+
+    git_remote_url = 'ssh://{user}@{host}/home/{user}/{code_dir}/.git'.format(
+        user=env.user,
+        host=env.host,
+        code_dir=env.offline_code_dir
+    )
+    local('cd {}/commcare-hq && git push {} master'.format(
+        OFFLINE_STAGING_DIR,
+        git_remote_url,
+        env.deploy_metadata.deploy_ref,
+    ))
+
+    update_bower_offline()
+    update_npm_offline()
+
+
+def clone_current_release_to_home_directory():
+    _clone_code_from_local_path(env.code_current, env.offline_code_dir)
+
+
+def update_bower_offline():
+    _upload_and_extract(os.path.join(
+        OFFLINE_STAGING_DIR, BOWER_ZIP_NAME
+    ))
+
+
+def update_npm_offline():
+    _upload_and_extract(os.path.join(
+        OFFLINE_STAGING_DIR, NPM_ZIP_NAME
+    ))
+
+
+def upload_pip_wheels():
+    _upload_and_extract(os.path.join(
+        OFFLINE_STAGING_DIR, WHEELS_ZIP_NAME
+    ))
+
+
+def _upload_and_extract(zipname):
+    put(zipname, env.offline_code_dir)
+    run('unzip {}/{}'.format(env.offline_code_dir, zipname))
```

Comment:
We don't have unzip on softlayer. I ran into that the other day uploading certs. Do we have it on NIC? May be better to use the python zip library, or tar.gz/tar.bz2